src/code/fv-components/egg-hatch-drake.js

```diff
@@ -0,0 +1,53 @@
+import React, { PropTypes } from 'react';
+import AnimatedSprite from '../components/animated-sprite';
+import OrganismView from '../components/organism';
+import classNames from 'classnames';
+
+export default class EggHatchDrakeView extends React.Component {
+
+  static propTypes = {
+    id: PropTypes.string,
+    eggClasses: PropTypes.string,
+    drakeClasses: PropTypes.string,
+    drake: PropTypes.object,
+    hatchStarted: PropTypes.bool,
+    hatchComplete: PropTypes.bool,
+    onHatchComplete: PropTypes.func
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
Randomly playing with generating errors to test Rollbar, and I came across an error here and found this thread.

If a user is at #/6/1/5 and navigates to #/6/1/6, an error is thrown, because the egg hatching animation starts on the navigation event, finishes, and then errors because #/6/1/6 does not contain a target drake to check against.

Obviously, the egg hatching animation should not happen on the navigation event, but it does because #6/1/6 shows the drake, so `hatchStarted` gets set to true in this same view, and in the `componentWillReceiveProps` function nothing notices that `skipHatchAnimation` is also true. So the animation starts.

The simple way to fix this is to look at `skipHatchAnimation`:
```
componentWillReceiveProps(nextProps) {
    if (this.state.hatchStarted !== nextProps.hatchStarted)
      this.setState({ hatchStarted: nextProps.hatchStarted, hatchComplete: nextProps.skipHatchAnimation || false });
  }
```
However, without having looked too closely at this code or the conversations around it, I wonder if it wouldn't just be easier to have a `showDrake` or something property, that can skip all this logic. It seems strange to have to set two properties, `hatchStarted=true, skipHatchAnimation=true` when what you really mean is "just show the drake and pretend the egg doesn't exist," or `showDrake=true`.